incubator/mariadb/maria.libsonnet

```diff
@@ -0,0 +1,321 @@
+local k = import 'ksonnet.beta.2/k.libsonnet';
+local deployment = k.extensions.v1beta1.deployment;
+
+{
+  parts:: {
+
+    svc(namespace, name, metricsEnabled=true, labels={app:name}, selector={app:name})::
+      local defaults = {
+        serviceType: "ClusterIP"
+        // i could totally hardcode this and skip the defaults object
+      };
+      {
+        apiVersion: "v1",
+        kind: "Service",
+        metadata: {
+          name: name,
+          labels: labels,
+          [if metricsEnabled then "annotations"]: {
+            "prometheus.io/scrape": "true",
+            "prometheus.io/port": "9104",
+          }
+        },
+        spec: {
+          type: defaults.serviceType,
+          ports: [
+            {
+              name: "mysql",
+              port: 3306,
+              targetPort: "mysql",
+            },
+          ] + if metricsEnabled then [
+            {
+              name: "metrics",
+              port: 9104,
+              targetPort: "metrics",
+            },
+          ] else [],
+          selector: selector,
+        },
+      },
+
+    secret(namespace, name, mariaRootPassword, mariadbPassword, labels={app:name},):: {
+      apiVersion: "v1",
+      kind: "Secret",
+      metadata: {
+        name: name,
+        namespace: namespace,
+        labels: labels,
+      },
+      type: "Opaque",
+      data: {
+         "mariadb-root-password":
+         if mariaRootPassword != null then std.base64(mariaRootPassword)
+          else error "Please add a root password",
+        "mariadb-password":
+        if mariadbPassword != null then std.base64(mariadbPassword)
+          else error "Please add a password",
+      },
+    },
+    configMap(namespace, name, labels={app:name})::
+      local config = |||
+        [mysqld]
+        innodb_buffer_pool_size=2G
+      |||;
+      {
+        apiVersion: "v1",
+        kind: "ConfigMap",
+        metadata: {
+          name: name,
+          namespace: namespace,
+          labels: labels,
+          },
+        data: {
+          "my.cnf": config,
+        },
+      },
+
+    pvc(namespace, name, storageClassName="-", labels={app:name})::
+      local defaults = {
+        accessMode: "ReadWriteOnce",
+        size: "8Gi"
+      };
+      {
+        kind: "PersistentVolumeClaim",
+        apiVersion: "v1",
+        metadata: {
+          name: name,
+          namespace: namespace,
+          labels: labels,
+        },
+        spec: {
+          accessModes: [
+            defaults.accessMode,
+          ],
+          resources: {
+            requests: {
+              storage: defaults.size,
+            },
+          },
+          [if storageClassName != null then "storageClass"]:storageClassName,
+        },
+      },
+    deployment:: {
```

Comment:
Missing newline between `pvc` and `deployment`

Reply:
depressing checkmark